src/GraphQL.Conventions/Extensions/GraphQLBuilderExtensions.cs

```diff
@@ -0,0 +1,65 @@
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using GraphQL.DI;
+
+namespace GraphQL.Conventions
+{
+    public static class GraphQLBuilderExtensions
+    {
+        public static IGraphQLBuilder AddConventionsSchema(this IGraphQLBuilder builder, Func<GraphQLEngine> factory)
+            => AddConventionsSchema(builder, _ => factory());
+
+        public static IGraphQLBuilder AddConventionsSchema(this IGraphQLBuilder builder, Func<IServiceProvider, GraphQLEngine> factory)
+        {
+            return builder.AddSchema(provider =>
+            {
+                var engine = factory(provider);
+                var schema = engine.GetSchema();
+                var schemaConfigurations = provider.GetService<IEnumerable<IConfigureSchema>>();
```

Comment:
Why do you run configurations here? They normally run from `Schema`'s ctor.

Reply:
Because when the Schema is constructed at GraphTypeAdapter.cs:47, the service provider passed to it is a reference to `DeriveTypeFromTypeInfo` at GraphTypeAdapter.cs:72, which returns `null` for interfaces, including `IEnumerable<IConfigureSchema>`.  So the schema constructor will never run the registered configurations [unless it is done here].

In general, the "old code" (the GraphQLEngine code that constructs a schema) is unaware of the DI changes to GraphQL.NET.  I've not attempted to update the old code, but simply to write a wrapper to make it compatible with the `IGraphQLBuilder`.

I'll write a test to ensure that it runs `ConfigureSchema` executions once and only once.

Reply:
> I'll write a test to ensure that it runs `ConfigureSchema` executions once and only once.

Done in #256

Reply:
🤔 

> which returns null for interfaces, the "old code" (the GraphQLEngine code that constructs a schema) is unaware of the DI changes to GraphQL.NET and is not based on DI.

Sounds not good. Is that by design?

Reply:
Probably just due to the design being originally based on GraphQL.NET 2.x.  I'd say this is up to @tlil to change as he desires.  I was just trying to get the packages compatible with `IGraphQLBuilder` and eliminate duplicated code for serialization and hosting.  Since I've added the test, now if it is changed, it will be apparent within the test.

Reply:
ok